datasources/src/main/scala/com/normation/plugins/datasources/Repository.scala

```diff
@@ -297,10 +298,16 @@ class DataSourceRepoImpl(
   /*
    * delete need to clean existing live resource
    */
-  override def delete(id: DataSourceId): IOResult[DataSourceId] = dataSourcesLock.withPermit {
+  override def delete(id: DataSourceId, cause: UpdateCause): IOResult[DataSourceId] = dataSourcesLock.withPermit {
     // start by cleaning
     datasources.delete(id) *>
-    backend.delete(id)
+    backend.delete(id, cause) *>
+    fetch.deleteAll(id, cause) *>
```

Comment:
I think we should delete the properties before deleting the datasources, or else if something goes wrong when deleting a property, the datasource is deleted and we can trigger properties deletion again. 

Maybe a clean up job at start up cleaning invalid datasource properties ? ( just thinking out loud)

Reply:
I think it would be more dangerous: we could be in a case where the datasource is there, but some properties related to it would be deleted (so the generation changed while the hypothesis didn't). On that order, at worst we remain in the same situation. 

Yes, for the cleaning, something is a broken right now. A bootstrap check seems a good first solution. I opened that ticket to track it: https://issues.rudder.io/issues/23679